source/particles/particle_handler.cc

```diff
@@ -919,15 +919,31 @@ namespace Particles
   IndexSet
   ParticleHandler<dim, spacedim>::locally_owned_particle_ids() const
   {
-    IndexSet set(get_next_free_particle_index());
+    IndexSet                           set(get_next_free_particle_index());
+    std::vector<types::particle_index> indices;
+    indices.reserve(n_locally_owned_particles());
     for (const auto &p : *this)
-      set.add_index(p.get_id());
+      indices.push_back(p.get_id());
+    set.add_indices(indices.begin(), indices.end());
     set.compress();
     return set;
   }
 
 
 
+  template <int dim, int spacedim>
+  types::particle_index
+  ParticleHandler<dim, spacedim>::get_max_local_particle_index() const
+  {
+    typename PropertyPool<dim, spacedim>::Handle max_index = 0;
+    for (const auto &pic : particles)
+      for (const auto numbers : pic)
+        max_index = std::max(numbers, max_index);
+    return max_index;
```

Comment:
Would the following be simpler? It could return slightly larger numbers than this loop (if there are empty slots at the end of the property pool), and the function would still need to be written in the PropertyPool class, but it could simply return `locations.size()`. This would save a loop over all particles.

```suggestion
    return property_pool->n_slots();
```

Reply:
This is an excellent idea. I have one question, though: Now we return the value one past the maximal value (which is the more appropriate one to set up vectors anyway), but I am not sure of `get_max_local_particle_index` is the right name any more. On the other hand, I cannot find a good alternative name either. Does anyone have another idea? If not, I think we can still stick with it as we new say that there can be "gaps".

Reply:
I think the name is fine. Maybe you could add a warning to the function documentation.

Reply:
I added a comment with how exactly this number looks like in `particle_handler.h`.